### PR TITLE
feat: [CI-3777]: search in tests

### PR DESCRIPTION
**Pull request by mtran7**
#### Screenshots

UPDATED Search Button with Searching on Enter Key Press. Search button was too prominent on this page and ux suggested to use blue text style

image (37)
With value filled in & button on hover
image

// Previous functionality is the same

https://user-images.githubusercontent.com/43422351/187319255-664ac645-8358-4154-87fe-7db3ec4f4e13.mov
image
Ungrouped View
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 5 Issues 
* Code issue 5 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10882)

**Comment by mtran7** [comment]
retrigger typecheck

**Review by vardanbansal-harness — DISMISSED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionTestView/TestsExecution.tsx [hunk 1]
```diff
@@ -355,6 +398,20 @@ export function TestsExecution({
...
+                <ExpandingSearchInput
+                  alwaysExpanded
+                  width={200}
+                  className={css.expandSearch}
+                  placeholder={getString('pipeline.testsReports.searchByTestName')}
+                />
+                <Button
+                  data-testid="search-btn"
+                  text="Search"
+                  onClick={searchTestNames}
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionTestView/TestsExecution.tsx:411]: debounce?

Comment (mtran7) [reply src/modules/70-pipeline/pages/execution/ExecutionTestView/TestsExecution.tsx:411]: mmm don't think it is necessary as user will explicitly press to fetch each time and it is not handled on keypress/enter

File: src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx [hunk 2]
```diff
@@ -215,6 +214,67 @@ describe('BuildTests snapshot test', () => {
...
+      >
+        <BuildTests />
+      </TestWrapper>
+    )
+
+    const chevronDownButton = container.querySelector('[class*="expanded"] [data-icon="chevron-down"]')
+
+    if (!chevronDownButton) {
+      throw Error('Cannot find chevron down button')
+    }
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx:248]: use findBy ?

File: src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx [hunk 3]
```diff
@@ -215,6 +214,67 @@ describe('BuildTests snapshot test', () => {
...
+    expect(document.querySelector('[class*="sortBySelect"]')).not.toBeNull()
+    expect(document.querySelector('[class*="activeGroupedIcon"] [data-icon="list-view"]')).toBeNull()
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx:250]: can we find better selectors than class names?

File: src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx [hunk 4]
```diff
@@ -215,6 +214,67 @@ describe('BuildTests snapshot test', () => {
...
+
+    const ungroupedButton = container.querySelector('[data-icon="list-view"]')
+    if (!ungroupedButton) {
+      throw Error('Cannot find ungrouped button')
+    }
+    fireEvent.click(ungroupedButton)
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution/ExecutionTestView/__test__/BuildTests.test.tsx:256]: userevent and then avoid the further waitFor?

**Review by udayvunnam — DISMISSED** [review]
Few comments in tests

**Pull request merged**